src/parquet/statistics.cc

```diff
@@ -107,8 +168,17 @@ void TypedRowGroupStatistics<DType>::Update(const T* values, int64_t num_not_nul
   // TODO: support distinct count?
   if (num_not_null == 0) return;
 
+  // PARQUET-1225: Handle NaNs
+  // The problem arises only if the starting/ending value(s)
```

Comment:
Do NaN-s at the end actually cause trouble? I had the impression that only NaN-s at the beginning are problematic.

Reply:
They are a problem even in the end. The max value becomes NaN. This is specific to the implementation of `minmax_element`. A possible implementation is specified here http://en.cppreference.com/w/cpp/algorithm/minmax_element